webdriver-spec.html

```diff
@@ -5512,6 +5522,33 @@ <h2>Element Interaction</h2>
  they will implicitly <a data-lt="scroll into view">scroll elements into view</a>
  and check that it is an <a>interactable element</a>.
 
+<p>Some <a>resettable elements</a> define their own <dfn>clear algorithm</dfn>.
+ Unlike their associated <a>reset algorithms</a>,
+ changes made to form controls as part of these algorithms
+ <em>does</em> count as changes caused by the user
+ (and thus, e.g. does cause <a><code>input</code></a> events to fire).
+ When the <a>clear algorithm</a> is invoked
+ for an element that does not define its own <a>clear algorithm</a>,
+ its <a>reset algorithm</a> must be invoked instead.
+
+<p>The <a>clear algorithm</a> for <a><code>input</code></a> elements
```

Comment:
This may be clearer as a switch. When I did the initial read, I though all cases applied in all cases. Maybe I need more coffee?

Reply:
I did first write this as a switch, but switches make it hard for other specifications to reference WebDriver. The other reason I chose this approach is that it is the same that HTML uses for ‘resettable elements’ and ‘reset algorithm’, which gives it a nice parity.